src/components/AddMovie.jsx

```diff
@@ -1 +1,72 @@
-// implement AddMovie component here
+import React, { Component } from 'react';
+
+class AddMovie extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      subtitle: '',
+      title: '',
+      imagePath: '',
+      storyline: '',
+      rating: 0,
+      genre: 'action',
+    };
+
+    this.handle = this.handle.bind(this);
+    this.includeMovie = this.includeMovie.bind(this);
+  }
+
+  resetState() {
+    this.setState(
+      {
+        subtitle: '',
+        title: '',
+        imagePath: '',
+        storyline: '',
+        rating: 0,
+        genre: 'action',
+      },
+    );
+  }
+
+  handle(e) {
+    this.setState({ [e.target.name]: e.target.value });
+  }
+
+  includeMovie() {
+    const { onClick } = this.props;
+    onClick(this.state);
+    this.resetState();
+  }
+
+  render() {
+    const { subtitle, title, imagePath, storyline, rating, genre } = this.state;
+    return (
+      <form>
+        <label htmlFor="title">Título
+          <input name="title" value={title} type="text" onChange={this.handle} /></label>
```

Comment:
Ai seguindo o que falei na função _handle_, a chamada ficaria assim:
```suggestion
          <input name="title" value={title} type="text" onChange={(event) => this.handle(event, 'title')}/></label>
```
Seguindo a mesma lógica para os outros elementos. Caso queira economizar linhas, você consegue fazer uma função genérica para criar esses elementos.